x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentToken.java

```diff
@@ -0,0 +1,244 @@
+
+    private static final Logger logger = LogManager.getLogger(CreateEnrollmentToken.class);
+    private final Environment environment;
+    private final SSLService sslService;
+    private final CommandLineHttpClient client;
+    private final URL defaultUrl;
+
+    public CreateEnrollmentToken(Environment environment) throws MalformedURLException {
+        this(environment, new CommandLineHttpClient(environment));
+    }
+
+    // protected for testing
+    protected CreateEnrollmentToken(Environment environment, CommandLineHttpClient client) throws MalformedURLException {
+        try {
+            this.environment = environment;
+            this.sslService = new SSLService(environment);
+            this.client = client;
+            this.defaultUrl = new URL(client.getDefaultURL());
+        } catch (MalformedURLException e) {
+            logger.error(("Error generating enrollment token"), e);
+            throw new MalformedURLException("Error generating enrollment token: " + e.getMessage());
+        }
+    }
+
+    public String createNodeEnrollmentToken(String user, SecureString password) throws Exception {
+        if (XPackSettings.ENROLLMENT_ENABLED.get(environment.settings()) != true) {
+            throw new IllegalStateException("[xpack.security.enrollment.enabled] must be set to `true` to create an enrollment token");
+        }
+        final String fingerprint = getCaFingerprint();
+        final String apiKey = getApiKey(user, password);
+        final Tuple<List<String>, String> httpInfo = getNodeInfo(user, password);
+
+        try {
+            final XContentBuilder builder = JsonXContent.contentBuilder();
+            builder.startObject();
+            builder.field("ver", httpInfo.v2());
+            builder.startArray("adr");
+            for (String bound_address : httpInfo.v1()) {
+                builder.value(bound_address);
+            }
+            builder.endArray();
+            builder.field("fgr", fingerprint);
+            builder.field("key", apiKey);
+            builder.endObject();
+            final String jsonString = Strings.toString(builder);
+            return Base64.getUrlEncoder().encodeToString(jsonString.getBytes(StandardCharsets.UTF_8));
+        } catch (Exception e) {
+            logger.error(("Error generating enrollment token"), e);
+            throw new IllegalStateException("Error generating enrollment token: " + e.getMessage());
+        }
+    }
+
+    private HttpResponse.HttpResponseBuilder responseBuilder(InputStream is) throws IOException {
+        final HttpResponse.HttpResponseBuilder httpResponseBuilder = new HttpResponse.HttpResponseBuilder();
+        if (is != null) {
+            byte[] bytes = toByteArray(is);
+            String responseBody = new String(bytes, StandardCharsets.UTF_8);
+            logger.debug(responseBody);
+            httpResponseBuilder.withResponseBody(responseBody);
+        } else {
+            logger.debug("<Empty response>");
```

Comment:
`is` cannot be `null`.
Even if it could, that is an error condition, not a valid empty response.